chia/types/generator_types.py

```diff
@@ -34,10 +25,7 @@ class CompressorArg:
 @streamable
 class BlockGenerator(Streamable):
     program: SerializedProgram
-    generator_args: List[GeneratorArg]
-
-    def block_height_list(self) -> List[uint32]:
-        return [a.block_height for a in self.generator_args]
+    generator_refs: List[SerializedProgram]
 
-    def generator_refs(self) -> List[SerializedProgram]:
-        return [a.generator for a in self.generator_args]
+    # the heights are only used when creating blocks, never when validating
+    block_height_list: List[uint32]
```

Comment:
this is the main change. Instead of member functions generating those lists, make them members directly. This alters the syntax for accessing them from a function call to a property access, which is the main reason for the other changes in this PR